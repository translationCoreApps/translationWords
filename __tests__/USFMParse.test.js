/* eslint-env jest */
import usfmToJSON from '../src/USFMParse';
import fs from 'fs-extra';
const titus_usfm_file = '__tests__/fixtures/project/57-TIT.usfm';

describe('USFMParse Tests', () => {
  it('Test usfmToJSON', () => {
    const usfm = fs.readFileSync(titus_usfm_file, 'utf8');
    const json = usfmToJSON(usfm);
    expect(json).toMatchSnapshot();
  });
});