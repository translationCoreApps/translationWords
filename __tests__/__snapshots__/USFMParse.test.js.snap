// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`USFMParse Tests Test usfmToJSON 1`] = `
Object {
  "bookAbbr": "???",
  "chapters": Array [
    Object {
      "num": -1,
      "verses": Array [
        Object {
          "num": -1,
          "text": "TIT Unlocked Literal Bible
UTF-8
Titus
The Letter of Paul to Titus
Titus
Tit
Titus

",
        },
      ],
    },
    Object {
      "num": 1,
      "verses": Array [
        Object {
          "num": 1,
          "text": "Paul, a servant of God and an apostle of Jesus Christ, for the faith of God's chosen people and the knowledge of the truth that agrees with godliness,
",
        },
        Object {
          "num": 2,
          "text": "with the certain hope of everlasting life that God, who does not lie, promised before all the ages of time.
",
        },
        Object {
          "num": 3,
          "text": "At the right time, he revealed his word by the message that he trusted me to deliver, by the command of God our savior.

",
        },
        Object {
          "num": 4,
          "text": "To Titus, a true son in our common faith. Grace and peace from God the Father and Christ Jesus our savior.
",
        },
        Object {
          "num": 5,
          "text": "For this purpose I left you in Crete, that you might set in order things not yet complete and ordain elders in every city as I directed you.

",
        },
        Object {
          "num": 6,
          "text": "An elder must be without blame, the husband of one wife, with faithful children not accused of reckless behavior or undisciplined.
",
        },
        Object {
          "num": 7,
          "text": "It is necessary for the overseer, as God's household manager, to be blameless. He must not be arrogant, not be easily angered, not addicted to wine, not a brawler, and not a greedy man.

",
        },
        Object {
          "num": 8,
          "text": "Instead, he should be hospitable and a friend of what is good. He must be sensible, righteous, godly, and self-controlled.
",
        },
        Object {
          "num": 9,
          "text": "He should hold tightly to the trustworthy message that was taught, so that he may be able to encourage others with good teaching and correct those who oppose him.

",
        },
        Object {
          "num": 10,
          "text": "For there are many rebellious people, empty talkers and deceivers, especially those of the circumcision.
",
        },
        Object {
          "num": 11,
          "text": "It is necessary to stop them. They are upsetting whole families by teaching for shameful profit what they should not teach.

",
        },
        Object {
          "num": 12,
          "text": "One of their own prophets has said, \\"Cretans are always liars, evil beasts, lazy gluttons.\\"
",
        },
        Object {
          "num": 13,
          "text": "This statement is true. Therefore, correct them severely, so that they may be sound in the faith,
",
        },
        Object {
          "num": 14,
          "text": "not paying any attention to Jewish myths or to the commands of people who turn away from the truth.

",
        },
        Object {
          "num": 15,
          "text": "To those who are pure, all things are pure. But to those who are corrupt and unbelieving, nothing is pure, but both their minds and their consciences have been corrupted.
",
        },
        Object {
          "num": 16,
          "text": "They profess to know God, but they deny him by their actions. They are detestable, disobedient, and unfit for doing any good work.

",
        },
      ],
    },
    Object {
      "num": 2,
      "verses": Array [
        Object {
          "num": 1,
          "text": "But you, speak what fits with faithful instruction.
",
        },
        Object {
          "num": 2,
          "text": "Teach older men to be temperate, dignified, sensible, sound in faith, in love, and in perseverance.

",
        },
        Object {
          "num": 3,
          "text": "Teach older women likewise to be reverent in behavior, not slanderers or being slaves to much wine, but to be teachers of what is good.
",
        },
        Object {
          "num": 4,
          "text": "In this way they may train the younger women to love their own husbands and children,
",
        },
        Object {
          "num": 5,
          "text": "to be sensible, pure, good housekeepers, and obedient to their own husbands, so that God's word may not be insulted.

",
        },
        Object {
          "num": 6,
          "text": "In the same way, encourage the younger men to use good sense.
",
        },
        Object {
          "num": 7,
          "text": "In all ways present yourself as an example of good works. In your teaching, show integrity, dignity,
",
        },
        Object {
          "num": 8,
          "text": "and a correct message that is above criticism, so that anyone who opposes you may be ashamed because they have nothing bad to say about us.

",
        },
        Object {
          "num": 9,
          "text": "Teach slaves to obey their masters in everything, to please them and not argue with them,
",
        },
        Object {
          "num": 10,
          "text": "to not steal from them, but instead to demonstrate all good faith, so that in every way they may bring credit to the teaching about God our Savior.

",
        },
        Object {
          "num": 11,
          "text": "For the grace of God has appeared for the salvation of all people.
",
        },
        Object {
          "num": 12,
          "text": "It trains us to reject godlessness and worldly passions, and to live self-controlled, upright, and godly lives in this age,
",
        },
        Object {
          "num": 13,
          "text": "while we look forward to receiving our blessed hope, the appearance of the glory of our great God and Savior Jesus Christ.

",
        },
        Object {
          "num": 14,
          "text": "Jesus gave himself for us in order to redeem us from all lawlessness and to make pure, for himself, a special people who are eager to do good works.

",
        },
        Object {
          "num": 15,
          "text": "Speak of these things, encourage people to do them, and give correction with all authority. Let no one disregard you.

",
        },
      ],
    },
    Object {
      "num": 3,
      "verses": Array [
        Object {
          "num": 1,
          "text": "Remind them to submit to rulers and authorities, to obey them, to be ready for every good work,
",
        },
        Object {
          "num": 2,
          "text": "to revile no one, to not be eager to fight, and to be gentle, showing all humility toward everyone.

",
        },
        Object {
          "num": 3,
          "text": "For once we ourselves were thoughtless and disobedient. We were led astray and enslaved by various passions and pleasures. We lived in evil and envy. We were detestable and hated one another.

",
        },
        Object {
          "num": 4,
          "text": "But when the kindness of God our savior and his love for mankind appeared,
",
        },
        Object {
          "num": 5,
          "text": "it was not by works of righteousness that we did, but by his mercy that he saved us, through the washing of new birth and renewal by the Holy Spirit,

",
        },
        Object {
          "num": 6,
          "text": "whom God richly poured on us through our Savior Jesus Christ,
",
        },
        Object {
          "num": 7,
          "text": "so that having been justified by his grace, we might become heirs with the certain hope of eternal life.

",
        },
        Object {
          "num": 8,
          "text": "This message is trustworthy. I want you to insist on these things, so that those who have believed in God may be careful to engage themselves in good works. These things are good and useful for everyone.

",
        },
        Object {
          "num": 9,
          "text": "But avoid foolish debates and genealogies and strife and conflict about the law. Those things are unprofitable and worthless.
",
        },
        Object {
          "num": 10,
          "text": "Reject anyone who is causing divisions among you, after one or two warnings,
",
        },
        Object {
          "num": 11,
          "text": "knowing that such a person has turned from the right way and is sinning and condemns himself.

",
        },
        Object {
          "num": 12,
          "text": "When I send Artemas or Tychicus to you, hurry and come to me at Nicopolis, where I have decided to spend the winter.
",
        },
        Object {
          "num": 13,
          "text": "Do everything you can to send on their way Zenas the lawyer and Apollos, so that they lack nothing. 

",
        },
        Object {
          "num": 14,
          "text": "Our people must learn to engage themselves in good works that provide for urgent needs, and so not be unfruitful.

",
        },
        Object {
          "num": 15,
          "text": "All those who are with me greet you. Greet those who love us in faith. Grace be with all of you.
",
        },
      ],
    },
  ],
}
`;
